superduperdb/misc/superduper.py

```diff
@@ -1,7 +1,11 @@
+from __future__ import annotations
+
 import typing as t
 
 from sklearn.linear_model._base import LinearRegression
-from torch.nn.modules.linear import Linear
+
+if t.TYPE_CHECKING:
```

Comment:
This won't actually work on client development systems if they use `mypy`.

When they import our code in `misc/superduper.py` during the type checking phase, then it will try to import torch, and fail.

`auto_identify` is well-defined for _any_ object, why not get rid of the arg declaration altogether and make it `t.Any`?

Reply:
But that's what the development dependencies are for, right? Here we are only concerned with users of our package, not developers.

(But I completely take your point about `auto_identify` - fixed!)